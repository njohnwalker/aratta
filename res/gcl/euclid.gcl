@req(true)
x, y := m, n ;
if x < 0 -> x := 0 - x ;
[] y < 0 -> y := 0 - y ;
fi
do @inv()
[]  x > y -> x   := x - y ;
[]  y > x -> x, y := y, x ;
od
// cannot express GCD postcondition with this logic
@ens( x == y
    & existsJ * x == m
    & existsK * x == n
    )
