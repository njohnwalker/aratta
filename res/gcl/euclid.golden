@req(true)
x, y := m, n ;
if 0 > x ->
      x := 0 - x ;
[] 0 > y ->
      y := 0 - y ;
fi
do @inv()
[] x > y ->
      x := x - y ;
[] y > x ->
      x, y := y, x ;
od
@ens((x == y) & (((existsJ * x) == m) & ((existsK * x) ==
n)))
